pkg/reconciler/events/auditlogs/auditlogs.go

```diff
@@ -61,121 +55,42 @@ type Reconciler struct {
-	// overwrite a prior update to status with this stale state.
-	if !equality.Semantic.DeepEqual(original.Status, s.Status) {
-		if err := c.updateStatus(ctx, original, s); err != nil {
-			c.Logger.Warn("Failed to update CloudAuditLogsSource status", zap.Error(err))
-			c.Recorder.Eventf(s, corev1.EventTypeWarning, "UpdateFailed",
-				"Failed to update status for CloudAuditLogsSource %q: %v", s.Name, err)
-			return err
-		} else if reconcileErr == nil {
-			c.Recorder.Eventf(s, corev1.EventTypeNormal, "Updated", "Updated CloudAuditLogsSource %q", s.Name)
-		}
-	}
-
-	if reconcileErr != nil {
-		c.Recorder.Event(s, corev1.EventTypeWarning, "InternalError", reconcileErr.Error())
-	}
-	return reconcileErr
-}
-
-func (c *Reconciler) reconcile(ctx context.Context, s *v1alpha1.CloudAuditLogsSource) error {
+func (c *Reconciler) ReconcileKind(ctx context.Context, s *v1alpha1.CloudAuditLogsSource) reconciler.Event {
 	ctx = logging.WithLogger(ctx, c.Logger.With(zap.Any("auditlogsource", s)))
 
 	s.Status.InitializeConditions()
+	s.Status.ObservedGeneration = s.Generation
 
-	// See if the source has been deleted.
-	if s.DeletionTimestamp != nil {
-		if err := c.deleteSink(ctx, s); err != nil {
-			s.Status.MarkSinkNotReady("SinkDeleteFailed", "Failed to delete Stackdriver sink: %s", err.Error())
-			return err
-		}
-		s.Status.MarkSinkNotReady("SinkDeleted", "Successfully deleted Stackdriver sink: %s", s.Status.StackdriverSink)
-
-		if err := c.PubSubBase.DeletePubSub(ctx, s); err != nil {
-			return err
-		}
-		s.Status.StackdriverSink = ""
-		c.removeFinalizer(s)
-		return nil
-	}
-
-	// Ensure the finalizer's there, since we're about to attempt
-	// to change external state with the topic, so we need to
-	// clean it up.
-	c.addFinalizer(s)
 	topic := resources.GenerateTopicName(s)
 	t, ps, err := c.PubSubBase.ReconcilePubSub(ctx, s, topic, resourceGroup)
 	if err != nil {
-		return err
+		return reconciler.NewEvent(corev1.EventTypeWarning, reconciledPubSubFailedReason, "Reconcile PubSub failed with: %s", err.Error())
 	}
 	c.Logger.Debugf("Reconciled: PubSub: %+v PullSubscription: %+v", t, ps)
 
 	sink, err := c.reconcileSink(ctx, s)
 	if err != nil {
-		return err
+		return reconciler.NewEvent(corev1.EventTypeWarning, reconciledFailedReason, "Reconcile Sink failed with: %s", err.Error())
 	}
 	s.Status.StackdriverSink = sink
 	s.Status.MarkSinkReady()
 	c.Logger.Debugf("Reconciled Stackdriver sink: %+v", sink)
 
-	return nil
+	return reconciler.NewEvent(corev1.EventTypeNormal, reconciledSuccessReason, `CloudAuditLogsSource reconciled: "%s/%s"`, s.Namespace, s.Name)
 }
 
 func (c *Reconciler) reconcileSink(ctx context.Context, s *v1alpha1.CloudAuditLogsSource) (string, error) {
 	sink, err := c.ensureSinkCreated(ctx, s)
 	if err != nil {
-		s.Status.MarkSinkNotReady("SinkCreateFailed", "failed to ensure creation of logging sink: %v", err)
+		s.Status.MarkSinkNotReady("SinkCreateFailed", "failed to ensure creation of logging sink: %v", err.Error())
```

Comment:
`%s` if you do `err.Error()`, 
same below

Reply:
let's get this in so that we unblock @grac3gao ...
you can update in a follow up, no big deal

Reply:
@grac3gao same here... would you mind addressing this nit in your bigger PR?

Reply:
Yeah, I'll change it.